app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/OAuthRefreshTokenProcessor.java

```diff
@@ -126,6 +106,20 @@ void tryToRefreshAccessToken() {
                 final JsonNode accessToken = body.get("access_token");
                 if (accessToken != null && !accessToken.isNull() && !accessToken.asText().isEmpty()) {
                     component.setAccessToken(accessToken.asText());
+                    LOG.info("Successfull AuthToken Refresh");
+                    
+                    Long expiresInSeconds = null;
+                    if (expiresInOverride != null) {
+                        expiresInSeconds = Long.valueOf(expiresInOverride);
+                    } else {
+                        final JsonNode expiresIn = body.get("expires_in");
+                        if (expiresIn != null && !expiresIn.isNull() && !expiresIn.asText().isEmpty()) {
+                            expiresInSeconds = expiresIn.asLong();
+                        }
+                    }
+                    if (expiresInSeconds != null) {
+                        component.setExpiresAt(new Date(new Date().getTime() + expiresInSeconds * 1000 - 60000l));
```

Comment:
I'm not 100% sure so I steered away from mutating the component state, component is shared among endpoints, so if we have two API actions using it there might be a race condition. Could be perfectly OK, but I was thinking it might be safer to store state mutations in the processors.
Again it might not be so good at all if we have competing processors refreshing the access token at the same time.
I think it would be good to make the state mutation logic the same though -- so either mutate the component state or mutate the processor state.

Reply:
Also wouldn't it be easier just to store, a `long` or `AtomicLong` instead of `java.util.Date`, less objects to instantiate. And `System::currentTimeMillis` is in UTC, so just as safe as `java.util.Date`.